src/woo_publications/publications/tests/test_document_api_endpoints.py

```diff
@@ -2009,3 +2012,148 @@ def test_download_incomplete_document(self):
         response = self.client.get(endpoint, headers=AUDIT_HEADERS)
 
         self.assertEqual(response.status_code, status.HTTP_502_BAD_GATEWAY)
+
+
+@override_settings(ALLOWED_HOSTS=["testserver", "host.docker.internal"])
+class DocumentApiDeleteTests(VCRMixin, TokenAuthMixin, APITestCase):
+    DOCUMENT_TYPE_UUID = "9aeb7501-3f77-4f36-8c8f-d21f47c2d6e8"
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        # Set up global configuration
+        cls.service = service = ServiceFactory.create(
+            for_documents_api_docker_compose=True
+        )
+        config = GlobalConfiguration.get_solo()
+        config.documents_api_service = service
+        config.organisation_rsin = "000000000"
+        config.save()
+
+    def setUp(self):
+        super().setUp()
+        self.addCleanup(GlobalConfiguration.clear_cache)
+
+    @patch(
+        "woo_publications.contrib.documents_api.client.DocumentenClient.destroy_document"
+    )
+    def test_delete_document_no_document_connected(
+        self, mock_delete_document: MagicMock
+    ):
+        document = DocumentFactory.create(document_service=None, document_uuid=None)
+        url = reverse(
+            "api:document-detail",
+            kwargs={"uuid": str(document.uuid)},
+        )
+
+        response = self.client.delete(url, headers=AUDIT_HEADERS)
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        mock_delete_document.assert_not_called()
```

Comment:
This should work too, without a mock.
```suggestion
        self.assertEqueal(len(self.cassette), 0)
```
And it checks the (absence of) effect, not the implementation.

But do check the index removal task wasn't called.